display_list/testing/dl_test_surface_software.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/display_list/testing/dl_test_surface_software.h"
+
+#include "third_party/skia/include/core/SkCanvas.h"
+#include "third_party/skia/include/core/SkSurface.h"
+
+namespace flutter {
+namespace testing {
+
+using PixelFormat = DlSurfaceProvider::PixelFormat;
+
+bool DlSoftwareSurfaceProvider::InitializeSurface(size_t width,
+                                                  size_t height,
+                                                  PixelFormat format) {
+  primary_ = MakeOffscreenSurface(width, height, format);
+  return true;
```

Comment:
Maybe `!!primary_`? When would initialization ever fail?

Reply:
OOM? It doesn't have much configuration to get wrong with the SW provider.

Reply:
Done, but the others don't really test for success in creating the surface beyond validating the arguments and pixel format against their implementation capabilities...?